src/utils/index.rs

```diff
@@ -31,15 +31,29 @@ use std::fmt::Display;
 /// # Errors and Failures
 /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds) if
 /// either the index is negative or it does not fit into an [`i64`].
-pub fn evaluate_index<S: TryInto<i64> + Display + Copy>(index: S) -> Result<i64, MathError> {
+pub fn evaluate_index<S: TryInto<i64> + Display>(index: S) -> Result<i64, MathError> {
     // the index must fit into an [`i64`]
-    let index: i64 = match index.try_into() {
-        Ok(index) => index,
-        _ => {
-            return Err(MathError::OutOfBounds(
-                "fit into a i64".to_owned(),
-                index.to_string(),
-            ))
+
+    let index: i64 = if cfg!(debug_assertions) {
+        let index_str = index.to_string();
+        match index.try_into() {
+            Ok(index) => index,
+            _ => {
+                return Err(MathError::OutOfBounds(
+                    "fit into a i64".to_owned(),
+                    index_str,
+                ))
+            }
```

Comment:
it feel like this can be shorter, but I failed to find a better way. It would be nice if the tryInto returns a matherrror, if we would be able to propagate it

Reply:
Also didn't find a good way to do this :disappointed: